### PR TITLE
fix comment expansion

### PR DESCRIPTION
**Pull request by Raemon**

### CODE REVIEW

File: packages/lesswrong/components/comments/CommentsNode.jsx [hunk 1]
```diff
@@ -121,6 +121,7 @@ class CommentsNode extends Component {
       return true;
     
     return (
+      this.isTruncated() &&
       currentUser?.beta &&
       lowKarmaOrCondensed &&
       !(mostRecent && condensed) &&
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsNode.jsx]: When I do algebraic simplification on this line, I get
```suggestion
      (!condensed || lastCommentId === comment._id)
```
That doesn't make a whole lot of sense, but if it's wrong, it's at least stating the wrong thing more directly than it was before.

Comment (jimrandomh) [reply packages/lesswrong/components/comments/CommentsNode.jsx]: I guess the last two lines simplify to
```
  (condensed
    ? lastCommentId === comment._id
    : comment.baseScore < 10)
```

File: packages/lesswrong/components/comments/CommentsList.jsx [hunk 2]
```diff
@@ -70,7 +70,7 @@ class CommentsList extends Component {
           <div>
             {comments.map(comment =>
               <Components.CommentsNode
-                startThreadTruncated={startThreadTruncated || totalComments >= 50}
+                startThreadTruncated={startThreadTruncated || totalComments >= 100}
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsList.jsx]: Why?

File: packages/lesswrong/components/comments/CommentsNode.jsx [hunk 3]
```diff
@@ -184,9 +185,10 @@ class CommentsNode extends Component {
   }
 
   isTruncated = () => {
-    const { expandAllThreads } = this.props;
-    const { truncatedStateSet } = this.state
-    return !expandAllThreads && (this.state.truncated || (this.props.truncated && truncatedStateSet === false))
+    const { expandAllThreads, startThreadTruncated } = this.props;
+    // const { truncatedStateSet } = this.state
+    const truncatedStateSetFalse = !this.state || (this.state.truncatedStateSet === false)
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsNode.jsx]: `state` is never null, so this is just `!this.state.truncatedStateSet`.

File: packages/lesswrong/components/comments/CommentsNode.jsx [hunk 4]
```diff
@@ -184,9 +185,10 @@ class CommentsNode extends Component {
...
+    return !expandAllThreads && (this.state?.truncated || ((this.props.truncated && truncatedStateSetFalse) || (startThreadTruncated && truncatedStateSetFalse)))
```
Comment (jimrandomh) [inline packages/lesswrong/components/comments/CommentsNode.jsx]:
```suggestion
    if (expandAllThreads) return false;
    if (this.state.truncated) return true;
    if (!this.state.truncatedStateSet) return false;
    return this.props.truncated || startThreadTruncated;
```

File: packages/lesswrong/components/comments/CommentsList.jsx (hunk 2 above)
Comment (Raemon) [reply packages/lesswrong/components/comments/CommentsList.jsx]: it's the right number (mostly just due to experimentation and noticing that 50 comments was too early to do the thing)

**Comment by jimrandomh** [comment]
Approving on trust that you've tested it thoroughly. The code here is quite confusing, though.

**Pull request merged**